docs/fsharp/style-guide/conventions.md

```diff
@@ -513,13 +513,15 @@ The previous observations about performance with struct tuples and records also
     let reverseName (Name s) =
         s.ToCharArray()
         |> Array.rev
-        |> string
+        |> Array.map Char.ToString
+        |> String.concat ""
```

Comment:
This is certainly confusing, but it should actually be `System.String` and not `string`.

Explanation:

Pipelining into `System.String` calls the `String()` constructor, passing in the reverse `char[]`, which then creates a new string.

Pipelining into `string` passes the `char[]` to a function that calls `.ToString()` on that array. This produces what you saw.